error: 'isnan' was not declared in this scope

Hi all.

`OpenMS-2.8.0` is failing with `GCC-12.0.1`:

```
[ 92%] Building CXX object src/openms/CMakeFiles/OpenMS.dir/source/QC/MQEvidenceExporter.cpp.o
cd /builddir/build/BUILD/OpenMS-2.8.0/build/src/openms && /usr/bin/g++ -DBOOST_ALL_NO_LIB -DOpenMS_EXPORTS -DQT_CORE_LIB -DQT_NETWORK_LIB -DQT_NO_DEBUG -DQT_SQL_LIB -I/builddir/build/BUILD/OpenMS-2.8.0/build/src/openms -I/builddir/build/BUILD/OpenMS-2.8.0/src/openms -I/builddir/build/BUILD/OpenMS-2.8.0/build/src/openms/OpenMS_autogen/include -I/builddir/build/BUILD/OpenMS-2.8.0/src/openms/include -I/builddir/build/BUILD/OpenMS-2.8.0/build/src/openms/include -I/builddir/build/BUILD/OpenMS-2.8.0/src/openswathalgo/include -I/builddir/build/BUILD/OpenMS-2.8.0/build/src/openswathalgo/include -I/builddir/build/BUILD/OpenMS-2.8.0/src/openswathalgo/thirdparty/MIToolbox/include -I/builddir/build/BUILD/OpenMS-2.8.0/src/openswathalgo/thirdparty/MIToolbox/src -isystem /builddir/build/BUILD/OpenMS-2.8.0/src/openms/thirdparty/eol-bspline -isystem /builddir/build/BUILD/OpenMS-2.8.0/src/openms/thirdparty/IsoSpec -isystem /builddir/build/BUILD/OpenMS-2.8.0/src/openms/thirdparty/json -isystem /builddir/build/BUILD/OpenMS-2.8.0/src/openms/thirdparty/evergreen/src -isystem /usr/include/libsvm -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtCore -isystem /usr/lib64/qt5/mkspecs/linux-g++ -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtSql -isystem /usr/include/WildMagic -isystem /usr/include/eigen3 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection  -Wno-cpp -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64  -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Wall -Wextra -Wno-non-virtual-dtor -Wno-unknown-pragmas -Wno-long-long -Wno-unused-function -Wno-variadic-macros -fPIC -fopenmp -MD -MT src/openms/CMakeFiles/OpenMS.dir/source/QC/MQEvidenceExporter.cpp.o -MF CMakeFiles/OpenMS.dir/source/QC/MQEvidenceExporter.cpp.o.d -o CMakeFiles/OpenMS.dir/source/QC/MQEvidenceExporter.cpp.o -c /builddir/build/BUILD/OpenMS-2.8.0/src/openms/source/QC/MQEvidenceExporter.cpp
make[3]: Leaving directory '/builddir/build/BUILD/OpenMS-2.8.0/build'
In file included from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/KERNEL/MSExperiment.h:39,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/KERNEL/ConsensusMap.h:41,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/QC/MQEvidenceExporter.h:39,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/source/QC/MQEvidenceExporter.cpp:35:
/builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/KERNEL/AreaIterator.h:58:19: warning: 'template<class _Category, class _Tp, class _Distance, class _Pointer, class _Reference> struct std::iterator' is deprecated [-Wdeprecated-declarations]
   58 |       public std::iterator<std::forward_iterator_tag, ValueT>
      |                   ^~~~~~~~
In file included from /usr/include/c++/12/string:45,
                 from /usr/include/c++/12/bits/locale_classes.h:40,
                 from /usr/include/c++/12/bits/ios_base.h:41,
                 from /usr/include/c++/12/ios:42,
                 from /usr/include/c++/12/istream:38,
                 from /usr/include/c++/12/fstream:38,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/QC/MQEvidenceExporter.h:37:
/usr/include/c++/12/bits/stl_iterator_base_types.h:127:34: note: declared here
  127 |     struct _GLIBCXX17_DEPRECATED iterator
      |                                  ^~~~~~~~
/builddir/build/BUILD/OpenMS-2.8.0/src/openms/source/QC/MQEvidenceExporter.cpp: In member function 'void MQEvidence::exportRowFromFeature_(const OpenMS::Feature&, const OpenMS::ConsensusMap&, OpenMS::Size, const OpenMS::String&, const std::multimap<OpenMS::String, std::pair<long unsigned int, long unsigned int> >&, const OpenMS::ProteinIdentification::Mapping&)':
/builddir/build/BUILD/OpenMS-2.8.0/src/openms/source/QC/MQEvidenceExporter.cpp:318:7: error: 'isnan' was not declared in this scope
  318 |   if (isnan(uncalibrated_mz_error_ppm) && isnan(calibrated_mz_error_ppm))
      |       ^~~~~
/builddir/build/BUILD/OpenMS-2.8.0/src/openms/source/QC/MQEvidenceExporter.cpp:318:7: note: suggested alternatives:
In file included from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/DATASTRUCTURES/ListUtils.h:42,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/DATASTRUCTURES/DataValue.h:38,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/METADATA/MetaInfoInterface.h:41,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/KERNEL/RichPeak2D.h:38,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/KERNEL/BaseFeature.h:37,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/KERNEL/ConsensusFeature.h:38,
                 from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/include/OpenMS/QC/MQEvidenceExporter.h:38:
/usr/include/c++/12/cmath:632:5: note:   'std::isnan'
  632 |     isnan(_Tp __x)
      |     ^~~~~
In file included from /builddir/build/BUILD/OpenMS-2.8.0/src/openms/source/QC/MQEvidenceExporter.cpp:44:
/usr/include/boost/math/special_functions/fpclassify.hpp:605:14: note:   'boost::math::isnan'
  605 | inline bool (isnan)(T x)
      |              ^~~~~
make[3]: *** [src/openms/CMakeFiles/OpenMS.dir/build.make:9183: src/openms/CMakeFiles/OpenMS.dir/source/QC/MQEvidenceExporter.cpp.o] Error 1
```

--- a/src/openms/source/QC/MQEvidenceExporter.cpp
+++ b/src/openms/source/QC/MQEvidenceExporter.cpp
@@ -44,8 +44,6 @@
 #include <cmath> // isnan
 #include <fstream>
 
-#include <cmath> // isnan
-#include <fstream>
 
 using namespace OpenMS;
 

--- a/src/openms/source/QC/MQEvidenceExporter.cpp
+++ b/src/openms/source/QC/MQEvidenceExporter.cpp
@@ -41,10 +41,10 @@
 #include <OpenMS/SYSTEM/File.h>
 
 #include <QtCore/QDir>
-#include <boost/math/special_functions/fpclassify.hpp> // isnan
+#include <cmath> // isnan
 #include <fstream>
 
-#include <boost/math/special_functions/fpclassify.hpp> // isnan
+#include <cmath> // isnan
 #include <fstream>
 
 using namespace OpenMS;
@@ -315,7 +315,7 @@ void MQEvidence::exportRowFromFeature_(
   const double& uncalibrated_mz_error_ppm = ptr_best_hit->getMetaValue("uncalibrated_mz_error_ppm", NAN);
   const double& calibrated_mz_error_ppm = ptr_best_hit->getMetaValue("calibrated_mz_error_ppm", NAN);
 
-  if (isnan(uncalibrated_mz_error_ppm) && isnan(calibrated_mz_error_ppm))
+  if (std::isnan(uncalibrated_mz_error_ppm) && std::isnan(calibrated_mz_error_ppm))
   {
     file_ << "NA" << "\t"; // Mass error [ppm]
     file_ << "NA" << "\t"; // Uncalibrated Mass error [ppm]
@@ -324,7 +324,7 @@ void MQEvidence::exportRowFromFeature_(
     file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [ppm]
     file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [mDa]
   }
-  else if (isnan(calibrated_mz_error_ppm))
+  else if (std::isnan(calibrated_mz_error_ppm))
     {
       file_ << "NA" << "\t"; // Mass error [ppm]
       file_ << uncalibrated_mz_error_ppm << "\t"; // Uncalibrated Mass error [ppm]
@@ -333,7 +333,7 @@ void MQEvidence::exportRowFromFeature_(
       file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [ppm]
       file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [mDa]
     }
-    else if (isnan(uncalibrated_mz_error_ppm))
+    else if (std::isnan(uncalibrated_mz_error_ppm))
       {
         file_ << calibrated_mz_error_ppm << "\t"; // Mass error [ppm]
         file_ << "NA" << "\t"; // Uncalibrated Mass error [ppm]

